usmqe/api/tendrlapi/common.py

```diff
@@ -131,33 +132,60 @@ def get_job_attribute(self, job_id, attribute="status", section=None):
         else:
             return response.json()[attribute]
 
+    def get_job_messages(self, job_id):
+        """ Get messages from job specified by job_id.
+
+        Name:       "get_job_messages",
+        Method:     "GET",
+        Pattern     "jobs/:job_id:/messages",
+
+        Args:
+            job_id:     id of job
+        """
+        pattern = "jobs/{}/messages".format(job_id)
+        response = requests.get(
+            pytest.config.getini("usm_api_url") + pattern,
+            auth=self._auth,)
+        self.print_req_info(response)
+        self.check_response(response)
+        return response.json()
+
     def wait_for_job_status(
             self,
             job_id,
-            max_count=42,
+            job_time=10000,
+            update_time=1000,
             status="finished",
             issue=None,
-            sleep_time=5):
+            sleep_time=10):
         """ Repeatedly check if status of job with provided id is in required state.
+        It is time bounded by job_time and each event is time bounded by update_time
 
         Args:
             job_id: id provided by api request
-            max_count: maximum of iterations
+            job_time: job should achieve status in ``job_time`` seconds
+            update_time: event should be done in ``update_time`` seconds
             status: expected status of job that is checked
             issue: pytest issue message (usually github issue link)
             sleep_time: time in seconds between 2 job status function calls
         """
 
-        count = 0
+        start_time = datetime.datetime.now()
+        job_timeout = datetime.timedelta(0, job_time, 0)
+        update_timeout = datetime.timedelta(0, update_time, 0)
         current_status = ""
+        messages_count = 0
         while current_status not in (status, "finished", "failed") and\
-                count < max_count:
+            datetime.datetime.now() - start_time <= job_timeout and\
+                datetime.datetime.now() - start_time <= update_timeout:
```

Comment:
this check is not correct, the difference should not be from `start_time`, there has to be another variable e.g. `last_update` which will be equal to `start_time` at the beginning, and with any new message it will be reset to current time